syntax = "proto3";

package user;

service UserService {
    rpc Login(LoginRequest) returns (LoginResponse);
    rpc GetUser(GetUserRequest) returns (UserResponse);
}

message LoginRequest {
    string username = 1;
    string password = 2;
}

message LoginResponse {
    string user_id = 1;
    string token = 2;
    string message = 3;
}

message GetUserRequest {
    string user_id = 1;
}

message UserResponse {
    string user_id = 1;
    string username = 2;
    string email = 3;
}
